### PR TITLE
chore: add support for readonly arrays

### LINKED ISSUE #166: types: random.arrayElement (and probably others) should support ReadonlyArray
### Describe the bug

the new types in 6.0.0-alpha.3 (thanks!) don't handle cases where the DT types allowed `ReadonlyArray`, e.g. `random.arrayElement`.

### Reproduction

```ts
import faker from '@faker-js/faker'

const items = ['one', 'two', 'three'] as const

const pick = faker.random.arrayElement(items)
```

also note, that the other way around would be fine:

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtX1higE8BRCEAWxFQwB4AVAPgAosMqBnALngEEYRYoyYBKXgwBQksHk4Z4cKMDwRiAofAC88ANoByPCH0AaePowB3HKfMYAFnGMBdAp3izU86YRLkqNBgsSiqoahokotIA9NHwALSJ0qCQsAgo6Nh4iiDKquqCfhTUtCJsHJQ88ABKuaHhhcLM4vBSMnIKvsTaeoaoxmYW1rYWjiAu0iH5EWTFgSxdokA

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by clarkerican**
TODO: Revisit to add TS tests once this PR (https://github.com/faker-js/faker/pull/186/files) is merged. 

I searched for all parameters that have `T[]` and don't modify the array.

### CODE REVIEW

**Comment by Shinigami92** [comment]
Is this just enough? I expected something like `ReadonlyArray<T> | T[]` :thinking: 
Did you tested it locally? I'm afraid of breaking simple array typing ^^'

**Review by Shinigami92 — APPROVED** [review]
I tested a bit in [TS Playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAJwIZgCZwLYwF4CmAPACqIC8iAzlMjGAOYB8AFKsmgJ4BcKBqWMABtOiEgG0AuhUTiA5KjkAaRHIBGy1RDnTUVROADWYOAHckesVICUvMgG8AsAChEfKCGQWOqTuIAMkgDcLgC+Li4QCDSI7MgAjLw0dIxSMvKKKuqactqSEc5RYDHIBFQgQlCJ1LT0DDJomDj4BGwc8dYFRTFxAEy8AEr8giIAgj6cRMl1TOkKORpZeV3RUHzllf01KfWUjVi4hG3IvdZAA) and everything works find :+1:

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: src/helpers.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class Helpers {
    * @method faker.helpers.randomize
    * @param array
    */
-  randomize<T = string>(array: T[] = ['a', 'b', 'c'] as unknown as T[]): T {
+  randomize<T = string>(array: ReadonlyArray<T> = ['a', 'b', 'c'] as unknown as ReadonlyArray<T>): T {
```
[the remark this answers is not in the document]
Comment (stefanprobst) [reply src/helpers.ts:20]: why?

[the remark this answers is not in the document]
Comment (Shinigami92) [reply src/helpers.ts:20]: I'm with @stefanprobst, cause readonly arrays seem to just say that the function has the "contract" of: "I will not push or remove elements to/from this array you give me"
An `Array` is a subset of a `ReadonlyArray` and not the other way around, therefore an union of `ReadonlyArray<T> | T[]` would just collapse into `ReadonlyArray<T>` anyways.

**Review by ejcheng — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**